packages/nimble-components/src/patterns/button/styles.ts

```diff
@@ -78,26 +93,44 @@ export const styles = css`
     }
 
     .control:hover {
-        box-shadow: 0px 0px 0px ${borderWidth} ${borderHoverColor} inset;
-        outline: none;
+        border-color: ${borderHoverColor};
+        box-shadow: 0px 0px 0px 1px ${borderHoverColor} inset;
+        transition: box-shadow ${mediumDelay};
     }
 
     .control${focusVisible} {
+        border-color: ${borderHoverColor};
         box-shadow: 0px 0px 0px ${borderWidth} ${borderHoverColor} inset;
+    }
+
+    .control${focusVisible}::before {
```

Comment:
`::before` pseudoelements are treated as different elements in the DOM tree and should be organized in the CSS that way per our styling guidelines.
See: https://github.com/ni/nimble/blob/main/packages/nimble-components/docs/css-guidelines.md#group-selectors-by-target-and-document-order